Add datetime picker to set lobby start time

danxuliu opened the pull request:
![Talk-Moderation-Menu-Lobby-Timer-Picker](https://user-images.githubusercontent.com/26858233/66480203-5e56f400-ea9e-11e9-89d3-6fe55092be23.png)

danxuliu commented:
/backport to stable17

nickvergessen commented:
Please base your PR on stable17 and we merge it there for now, because of #vue

Ivansss commented:
I just noticed a rendering issue in the input field.

Chrome (macOS):
<img width="247" alt="Screenshot 2019-10-10 at 16 43 54" src="https://user-images.githubusercontent.com/4638605/66579902-d6481b80-eb7d-11e9-88fe-07c90cee71e4.png">
Firefox (macOS):
<img width="228" alt="Screenshot 2019-10-10 at 16 47 20" src="https://user-images.githubusercontent.com/4638605/66579932-dea05680-eb7d-11e9-9ddf-e3b09bcaeca3.png">
Safari (macOS):
<img width="235" alt="Screenshot 2019-10-10 at 16 48 31" src="https://user-images.githubusercontent.com/4638605/66579954-e65ffb00-eb7d-11e9-8225-e38920c0b1ec.png">

danxuliu commented:
Rebased on _stable17_, fixed the rendering issue (in my case it worked with Firefox but failed with Chromium on GNU/Linux... strange :-P ) and the width of the menu in narrow windows (please see the commit messages for details ;-) ).

Besides that I work arounded a bug with the datepicker pop up that was darkened when its menu item was not focused or hovered (again, details in the commit message ;-) ).

> the datepicker popover's shadow is very strong!

This is related to the darkened pop up. The reason is that there are two shadows, one for the main pop up and another one for the datepicker, because pop up menus use `filter: drop-shadow()`, so the shape of the parent pop up also includes the child. No, you can not remove the shadow of the child because then you have no shadow in its top, in the area that intersects with the parent pop up menu. No, you can not use `box-shadow` either, because then you lose the shadows in the arrows. No, I do not have more ideas, your turn :-P

Finally, I have fixed too the calendar icon in the menu item; it was there before, but not visible, although I am not sure if now it is too much also with the calendar icon in the date picker (which can not be removed from the component as far as I know, although maybe it could be hidden with CSS, I have not tried):
![Talk-Moderation-Menu-Lobby-Timer-Picker-With-Icon](https://user-images.githubusercontent.com/26858233/66662605-6b641680-ec49-11e9-893b-1c6de2156a97.png)

danxuliu commented:
>  Works fine, in my opinion the following 3 things should be changed:

I agree, but I think that they should be changed in the _DatetimePicker_ component of _nextcloud-vue_ instead of in Talk.

>  If you have the calendar popup open and click into the input field, the menu should close

I am not sure how to do this, as the [wrapped DatePicker](https://github.com/mengxiong10/vue2-datepicker) does not seem to provide a click event for the input field. Maybe using a native event and a reference to its internal input? In any case I am not sure if this should be the default behaviour or an option.

>  Instead of 0:00 the time should be set to either 12:00 or similar to the calendar app did before the migration to VueJS: to the next "reasonable time" (next full 15min. block I think it was)

If there is no date as soon as a day is chosen the value is updated, but that value does not contain hours nor minutes; this could be fixed by adding the next hour/minute block to the date whenever it is set and it was previously null (I guess that could be done with a _computedProperty_ or something like that). I think that that this would make sense as the default behaviour when the DatetimePicker type is _datetime_.

>  When you click on a minute block, the popup should close as well, as you are "done" serving input to the popup

I guess that this could be implemented by listening to change events, checking if only the minutes changed since the last known value and, in that case, calling [`showPanelNone()`](https://github.com/mengxiong10/vue2-datepicker/blob/a55b4b6f95a3300f4015409b8b292bb395dff6dc/src/calendar.vue#L438) on the wrapped `DatePicker`. Maybe this should be optional rather than a default behaviour, not sure.

Note that the DatetimePicker provides the _confirm_ property which shows an OK button that sets the date and closes the panel when clicked. However, as you then need to click on the arrow to save the setting I thought that it was better to immediately update the input with the date as soon as it was chosen in the calendar, without needing to confirm it.

@skjnldsv Thoughts? :-) (Mostly about whether this should go in the component itself rather than in Talk; you can ignore the implementation details for now ;-) ).

>  There is a small layout problem when you focus another window while the calendar popup is open

This is explained in [the message of the third commit](https://github.com/nextcloud/spreed/pull/2289/commits/acadbee0305f76870fa38b86a2fa6f3b4cd54579).

skjnldsv commented on js/views/vuewrapper.js:
```diff
@@ -0,0 +1,61 @@
+	/**
+	 * Helper class to wrap a Vue instance as a Marionette view.
+	 */
+	var VueWrapper = Marionette.View.extend({
```
There is laws against this! :scream:
